### ISSUE TITLE
Water renders incorrectly in the latest version

### ISSUE THREAD

**Issue by NotFrants** [misode/deepslate#43]
NBT Structure:
![](https://github.com/user-attachments/assets/8bf271d9-e3e2-4caa-8cda-1137d03aa9a2)

**Render in the latest version (0.23.2):**
![](https://github.com/user-attachments/assets/34578c5f-aadb-4739-a8fb-b095de614e2a)
Pixel size on the bottom & top face is exactly the block size (so its scaled 16 times), while the rest of the sides is 32px in width for some reason, so 0.5 block per pixel

**Same render in 0.22.4:**
![](https://github.com/user-attachments/assets/99c371a1-bb04-4440-a6ca-aa5f922f9b1b)

_Some more screenshots:_
Top:
![](https://github.com/user-attachments/assets/c6dd60f8-ba17-4344-b97e-9b7b5eef43a2)
Bottom:
![](https://github.com/user-attachments/assets/008e948f-ad55-475f-ab44-61921883ff19)
Side:
![](https://github.com/user-attachments/assets/4b0fa77c-5402-4d75-a682-c1f7a20b9f75)
Scale of the side pixels:
![](https://github.com/user-attachments/assets/bc736df4-9796-43a1-8a4e-e4193bbcf45c)

**Comment by NotFrants** [comment]
Same thing with lava:
![](https://github.com/user-attachments/assets/80128ee0-944f-4616-b356-a3ce613db8af)

**Comment by NotFrants** [comment]
I checked some more versions, the earliest the bug appears is in 0.23.0.

**Comment by NotFrants** [comment]
Wait nevermind, its 0.23.0-beta.1

**Comment by NotFrants** [comment]
Update: I found what's causing the bug. In the `/src/render/SpecialRenderer.ts`, there is a function called `getBlockMesh` (line 883). It first gets the model for the block, then scales it 1/16th (line 1012), and returns it.

However, at the start of the function, it checks if the block is water of lava, in which case it uses the `liquidRenderer` function,
and instead of saving the mesh into a variable, it instead returns it (lines 885 and 888), which makes it 16x what its supposed to be.

Here is the current code:
```javascript
if (block.is('water')) {
  return liquidRenderer('water', getInt(block, 'level'), atlas, cull, 0)
}
if (block.is('lava')) {
  return liquidRenderer('lava', getInt(block, 'level'), atlas, cull)
}
const mesh = new Mesh()
const chestRenderer = ChestRenderers.get(block.getName().toString())
```

And the correct code would be:
```javascript
const mesh = new Mesh()
if (block.is('water')) {
  const liquidMesh = liquidRenderer('water', getInt(block, 'level'), atlas, cull, 0)
}
if (block.is('lava')) {
  const liquidMesh = liquidRenderer('lava', getInt(block, 'level'), atlas, cull)
}
if (liquidMesh !== undefined) {
  mesh.merge(liquidMesh)
}

const chestRenderer = ChestRenderers.get(block.getName().toString())
```

Basically you create the mesh before checking if its a liquid, and then you merge the model with the mesh instead of returning it.
